MyOnboarding/src/MyOnboardingApp.Validation/Validators/InvariantValidator.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Linq;
+using MyOnboardingApp.Contracts.ErrorFactories;
+using MyOnboardingApp.Contracts.Validation;
+
+namespace MyOnboardingApp.Validation.Validators
+{
+    internal class InvariantValidator<T> : IInvariantValidator<T>
+        where T : class
+    {
+        private readonly IValidationCriterion<T>[] _criteria;
+        private readonly IErrorFactory _errorFactory;
+
+
+        public InvariantValidator(IValidationCriterion<T>[] criteria, IErrorFactory errorFactory)
+        {
+            _criteria = criteria;
+            _errorFactory = errorFactory;
+        }
+
+
+        public IItemWithErrors<T> Validate(T item)
+        {
+            var errors = _criteria
+                .SelectMany(criterion => criterion.Validate(item, _errorFactory))
+                .ToList()
+                .AsReadOnly();
```

Comment:
Shouldn't be needed (based on https://github.com/petrareb/kentico-onboarding-cs/pull/4#discussion_r263424985)

Reply:
Mind you, why do I want you to replace the `ToList().AsReadOnly()` with `ToArray()` (I really do) instead of just omitting both the `ToList` and `AsReadOnly` methods? 🙃

Reply:
Array == materialization && fixed size ✌